### PR TITLE
Implement the process for moving packages from testing to stable

### LINKED ISSUE #19: Add usage documentation
- [x] How to install opkg
- [x] How to add the toltec repository to opkg
- [x] How to update the list of packages
- [x] How to install, remove, upgrade packages
- [ ] Which channel to choose (stable/testing)

### LINKED ISSUE #20: Define a process for moving packages from testing to stable
* See [how Debian does it](https://wiki.debian.org/DebianTesting).
* See [how Arch Linux does it](https://wiki.archlinux.org/index.php/Official_repositories#Testing_repositories).

### PR DESCRIPTION
**Pull request by matteodelabre**
This creates the `docs/branches.md` document which explains what the two release branches are, which one users should choose, and how the transition of packages from one branch to another is handled.

It also updates `docs/package.md` to add related information about the `maintainer` field.

This implements the proposal in #20, and closes #20 as a consequence.

### CODE REVIEW

**Comment by matteodelabre** [comment]
This is mostly RFC, I tried to implement what we discussed, but please point out any problems in the documents.

**Comment by Eeems** [comment]
@matteodelabre If this is a RFC right now, might as well convert it to a draft until we decide it's ready.

**Comment by matteodelabre** [comment]
Not sure why the CI is failing since this only changes the docs.

**Comment by Eeems** [comment]
> GitHub Actions has encountered an internal error when running your job.

Probably related to the Microsoft outage that's going on right now: https://twitter.com/MSFT365Status/status/1310698087749283840

**Comment by Eeems** [comment]
Just reviewed the content and it looks good to me. It would be nice to set up automation as part of this PR to create the merge window PR on a schedule as well.

**Comment by raisjn** [comment]
what's the process of becoming a maintainer and are the expectations of maintainers outlined somewhere?

**Comment by matteodelabre** [comment]
> Just reviewed the content and it looks good to me. It would be nice to set up automation as part of this PR to create the merge window PR on a schedule as well.

Should this automatically-created PR contain all of the `package/` directory from the testing branch? I don’t think we can create empty PRs.

**Comment by matteodelabre** [comment]
> what's the process of becoming a maintainer and are the expectations of maintainers outlined somewhere?

The process by which a maintainer is assigned to a package is described in this PR. As the document says, they are expected to review PRs regarding the packages they maintain. The process for adding a new repository maintainer is not outlined here, and I’d say it’s out of the scope of this PR.

**Comment by Eeems** [comment]
> > Just reviewed the content and it looks good to me. It would be nice to set up automation as part of this PR to create the merge window PR on a schedule as well.
> 
> Should this automatically-created PR contain all of the `package/` directory from the testing branch? I don’t think we can create empty PRs.

I'd say no, likely we should just have a dummy file that we bump a timestamp in or something.

**Comment by matteodelabre** [comment]
It turns out [we can actually create empty PRs](https://github.com/toltec-dev/test-empty-prs/pull/1), by adding an empty commit to a branch (with `git commit --allow-empty`).

**Review by raisjn — DISMISSED** [review]

File: docs/branches.md
Comment (matteodelabre) [inline docs/branches.md]: Instead of only allowing to open a merge PR on Mondays with changes from the preceding Friday, we could open them on Saturdays and only merge starting from the following Monday. This would allow us to more easily review and test the changes over the week-end.

```suggestion
Each Saturday, a pull request can be opened with selected updates from the testing branch to merge in the stable branch.
This pull request can only be merged from the following Monday.
Each of these package changes must be tested by a maintainer different from the maintainer of the affected package.
```

File: docs/branches.md
Comment (matteodelabre) [inline docs/branches.md]: Some additional testing criteria that I suggest, following what we learned in the previous merge window.

```suggestion
1. It should work as intended by upstream.
2. It must not have any known major bugs.
3. It must not destroy any user data (e.g. from previous versions of the same package, from other packages, from the home directory).
4. It must not break other packages.
5. It must not lock the device in a state where it can not be used without troubleshooting from SSH.
```

Comment (raisjn) [reply docs/branches.md]: > 5. It must not lock the device in a state where it can not be used without troubleshooting from SSH.

just as a side note, i thought of another workaround last week for the issue with the device going into bad state: the device can usually be rebooted, so if the device gets into a bad state while away from your computer, you can still get out of the bad state by rebooting.

Comment (matteodelabre) [reply docs/branches.md]: That’s right. So maybe `5. It must not lock the device in a state where it can not be used without rebooting or troubleshooting from SSH.`?

Comment (raisjn) [reply docs/branches.md]: > 2. It must not have any known major bugs.

I'd like it if we addressed how fixes can be brought over to stable during the merge window. Based on the last merge, if a package of mine gets punted to the next merge, I'm unhappy compared to if a fix can be brought in during the testing window. I'm not sure if others feel this way

File: docs/branches.md
Comment (matteodelabre) [inline docs/branches.md:19]:
```suggestion
Each Saturday, a pull request can be opened with selected updates from the testing branch to merge in the stable branch.
Only bug fixes may be added to existing merge PRs over the weekend.
```

Comment (matteodelabre) [reply docs/branches.md:19]: To address concerns from @raisjnn in the conversation above (https://github.com/toltec-dev/toltec/pull/81#discussion_r502918648).

**Comment by matteodelabre** [comment]
I think it’s best if we move forward with stabilizing the actual merge process then implement automation in a future PR.

**Review by raisjn — DISMISSED** [review]
looks good to me. thanks for the update around bugfixes!

**Comment by matteodelabre** [comment]
Just needed to fix a link and add instructions for switching to `testing` before merging.

**Comment by matteodelabre** [comment]
If you still agree with this change set, could you re-approve the PR @raisjnn? Your review was dismissed since I changed the link in the README and added instructions to switch to `testing`. Thanks!

**Review by raisjn — APPROVED** [review]
thanks for adding the note about bug fixes!

**Pull request merged**
